### ISSUE TITLE
`testing-library/render-result-naming-convention` failing on file where testing-library is not even imported

### ISSUE THREAD

**Issue by azarbayejani** [testing-library/eslint-plugin-testing-library#445]
### Plugin version

v4.10.1

### ESLint version

v7.12.1

### Node.js version

12.x

### npm/yarn version

yarn 1.22.10

### Operating system

macOS Big Sur, version 11.5.1

### Bug description

The eslint rule `testing-library/render-result-naming-convention` is failing on a file that doesn't even import anything from `@testing-library`. Possibly due to another function named `render`?

For example, this file fails lint due to `testing-library/render-result-naming-convention` on line 4:

```js
import { render } from 'hamljs';

export default function renderHaml(renderText) {
  let returnText = renderText;
  ...
  return render(returnText.trim());
}
```

I expect 

### Steps to reproduce

Run yarn lint against given code sample:

```
import { render } from 'hamljs';

export default function renderHaml(renderText) {
  let returnText = renderText;
  
  return render(returnText.trim());
}
```

### Error output/screenshots

Failure:

```
error: `returnText` is not a recommended name for `render` returned value. Instead, you should destructure it, or name it using one of: `view`, `utils` (testing-library/render-result-naming-convention) at spec/webpack_assets/support/renderHaml.js:4:7:
  2 |
  3 | export default function renderHaml(renderText) {
> 4 |   let returnText = renderText;
```

### ESLint configuration

```
{
  "env": {
    "browser": true,
    "es6": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:jest/recommended",
    "plugin:jest/style",
    "plugin:testing-library/dom",
    "plugin:testing-library/react"
  ],
  "globals": {
    "$": "readonly",
    "Clipboard": "readonly",
    "given": "readonly"
  },
  "parser": "@babel/eslint-parser",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "sourceType": "module",
    "ecmaVersion": 2018
  },
  "plugins": [
    "react",
    "react-hooks",
    "jest",
    "testing-library"
  ],
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "rules": {
    "array-bracket-newline": [
      "error",
      "consistent"
    ],
    "array-bracket-spacing": "error",
    "array-callback-return": "error",
    "arrow-body-style": "error",
    "arrow-parens": [
      "error",
      "always"
    ],
    "arrow-spacing": "error",
    "block-scoped-var": "error",
    "block-spacing": "error",
    "brace-style": [
      2,
      "1tbs",
      {
        "allowSingleLine": true
      }
    ],
    "camelcase": "error",
    "comma-dangle": "error",
    "comma-spacing": "error",
    "comma-style": "error",
    "computed-property-spacing": "error",
    "consistent-this": [
      "error",
      "self"
    ],
    "dot-location": [
      "error",
      "property"
    ],
    "dot-notation": "error",
    "eol-last": "error",
    "eqeqeq": "error",
    "func-call-spacing": "error",
    "func-names": [
      "error",
      "never"
    ],
    "func-style": [
      "error",
      "declaration",
      {
        "allowArrowFunctions": true
      }
    ],
    "function-call-argument-newline": [
      "error",
      "consistent"
    ],
    "generator-star-spacing": "error",
    "guard-for-in": "error",
    "indent": [
      "error",
      2,
      {
        "SwitchCase": 1
      }
    ],
    "jest/consistent-test-it": "error",
    "jest/no-duplicate-hooks": "error",
    "jest/prefer-hooks-on-top": "error",
    "jest/prefer-spy-on": "error",
    "jest/require-top-level-describe": "error",
    "jest/valid-title": "error",
    "jsx-quotes": [
      "error",
      "prefer-single"
    ],
    "key-spacing": "error",
    "keyword-spacing": "error",
    "linebreak-style": [
      "error",
      "unix"
    ],
    "new-cap": "error",
    "new-parens": "error",
    "no-array-constructor": "error",
    "no-caller": "error",
    "no-confusing-arrow": "error",
    "no-else-return": "error",
    "no-eq-null": "error",
    "no-eval": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-label": "error",
    "no-fallthrough": "error",
    "no-implicit-coercion": "error",
    "no-implied-eval": "error",
    "no-iterator": "error",
    "no-labels": "error",
    "no-lone-blocks": "error",
    "no-lonely-if": "error",
    "no-loop-func": "error",
    "no-mixed-operators": "error",
    "no-multi-spaces": "error",
    "no-multiple-empty-lines": "error",
    "no-negated-condition": "error",
    "no-nested-ternary": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-wrappers": "error",
    "no-octal-escape": "error",
    "no-param-reassign": "error",
    "no-proto": "error",
    "no-restricted-imports": [
      "error",
      {
        "patterns": [
          "..*"
        ]
      }
    ],
    "no-restricted-syntax": [
      "error",
      {
        "selector": "ForOfStatement",
        "message": "for...of operators are not allowed as they are not supported in IE11 (even after transpilation)"
      },
      {
        "selector": "ArrayExpression > SpreadElement",
        "message": "Array spreads are not allowed as they are not supported in IE11 (even after transpilation)"
      }
    ],
    "no-return-assign": "error",
    "no-return-await": "error",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-shadow": "error",
    "no-throw-literal": "error",
    "no-trailing-spaces": "error",
    "no-undefined": "error",
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": "error",
    "no-unused-expressions": "error",
    "no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_",
        "varsIgnorePattern": "^_"
      }
    ],
    "no-useless-call": "error",
    "no-useless-computed-key": "error",
    "no-useless-concat": "error",
    "no-useless-constructor": "error",
    "no-var": "error",
    "no-void": "error",
    "no-whitespace-before-property": "error",
    "nonblock-statement-body-position": "error",
    "object-curly-spacing": [
      "error",
      "always"
    ],
    "object-shorthand": "error",
    "operator-assignment": "error",
    "padded-blocks": [
      "error",
      "never"
    ],
    "prefer-arrow-callback": [
      "error",
      {
        "allowNamedFunctions": true
      }
    ],
    "prefer-const": "error",
    "prefer-destructuring": [
      "error",
      {
        "array": false,
        "object": true
      }
    ],
    "prefer-object-spread": "error",
    "quote-props": [
      "error",
      "as-needed"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "react/boolean-prop-naming": "error",
    "react/button-has-type": "error",
    "react/no-array-index-key": "error",
    "react/no-deprecated": "error",
    "react/jsx-boolean-value": "error",
    "react/jsx-closing-bracket-location": "error",
    "react/jsx-closing-tag-location": "error",
    "react/jsx-curly-brace-presence": "error",
    "react/jsx-curly-newline": "error",
    "react/jsx-curly-spacing": [
      "error",
      {
        "when": "always",
        "children": true
      }
    ],
    "react/jsx-equals-spacing": "error",
    "react/jsx-first-prop-new-line": "error",
    "react/jsx-indent": [
      "error",
      2,
      {
        "checkAttributes": true,
        "indentLogicalExpressions": true
      }
    ],
    "react/jsx-key": "error",
    "react/jsx-no-comment-textnodes": "error",
    "react/jsx-no-duplicate-props": "error",
    "react/jsx-no-target-blank": "error",
    "react/jsx-no-undef": [
      "error"
    ],
    "react/jsx-pascal-case": [
      "error",
      {
        "ignore": [
          "DS"
        ]
      }
    ],
    "react/jsx-props-no-multi-spaces": "error",
    "react/jsx-space-before-closing": "error",
    "react/jsx-uses-react": "error",
    "react/jsx-uses-vars": "error",
    "react/react-in-jsx-scope": "error",
    "react-hooks/exhaustive-deps": [
      "error",
      {
        "additionalHooks": "useDeepCompareEffect"
      }
    ],
    "react-hooks/rules-of-hooks": "error",
    "rest-spread-spacing": "error",
    "semi": [
      "error",
      "always"
    ],
    "semi-spacing": "error",
    "semi-style": "error",
    "space-before-blocks": "error",
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "never",
        "named": "never",
        "asyncArrow": "always"
      }
    ],
    "space-in-parens": "error",
    "space-infix-ops": "error",
    "space-unary-ops": [
      "error",
      {
        "words": true,
        "nonwords": false
      }
    ],
    "spaced-comment": "error",
    "switch-colon-spacing": "error",
    "template-curly-spacing": "error",
    "template-tag-spacing": "error",
    "testing-library/no-unnecessary-act": "error",
    "testing-library/no-node-access": "off",
    "testing-library/prefer-screen-queries": "off",
    "unicode-bom": "error",
    "yield-star-spacing": "error",
    "yoda": "error"
  }
}
```

### Rule(s) affected

testing-library/render-result-naming-convention

### Anything else?

Note that this is fixed if I just rename the variable`renderText` to `haml`, for example.

### Do you want to submit a pull request to fix this bug?

No

**Comment by Belco90** [comment]
Hi @azarbayejani!

I'm afraid that const saving the returned value from `renderText` is being reported because of [`eslint-plugin-testing-library` Aggressive Reporting](https://github.com/testing-library/eslint-plugin-testing-library#aggressive-reporting). 

There are different ways of addressing this, you can check that in [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#errors-reported-in-non-testing-files).

Could you check one of those (the one that fits better for you) and let us know if it worked?

**Comment by azarbayejani** [comment]
I also expected that to work, but it did not!

It's complaining about this assignment:

```
const returnText = renderText;

**Comment by azarbayejani** [comment]
Ok, turning off `testing-library/custom-queries` also fixes this. I thought Aggressive Reporting was only going to happen if I called a function called `renderText`, not just if I reassigned it.

I ended up just renaming the variable.

**Comment by Belco90** [comment]
If that `renderHaml` function is not placed in a testing file, then I'd suggest enabling `eslint-plugin-testing-library` only on testing files through `overrides` ESLint config key. That way, you make sure the plugin will lint just code from testing files.
